klog_test.go

```diff
@@ -1493,6 +1597,63 @@ func (l *testLogr) WithValues(...interface{}) logr.Logger {
 	panic("not implemented")
 }
 
+type callDepthTestLogr struct {
+	testLogr
+	callDepth int
+}
+
+func (l *callDepthTestLogr) resetCallDepth() {
+	l.mutex.Lock()
+	defer l.mutex.Unlock()
+	l.callDepth = -1
+}
+
+func (l *callDepthTestLogr) WithCallDepth(depth int) logr.Logger {
+	l.mutex.Lock()
+	defer l.mutex.Unlock()
+	l.callDepth = depth
+	return l
+}
+
+func (l *callDepthTestLogr) Info(msg string, keysAndValues ...interface{}) {
+	l.mutex.Lock()
+	defer l.mutex.Unlock()
+	// Add 2 to depth for the wrapper function caller and for invocation in
+	// test case.
+	_, file, line, _ := runtime.Caller(l.callDepth + 2)
```

Comment:
Can you add a test where `Info` is called directly, without going through klog? That will verify that this offset here is correct.

Without verifying that, both the depth calculation in klog and here might be off by one (or more) and tests only pass because the offsets cancel themselves out.